src/typechecker/FStar.TypeChecker.TcTerm.fs

```diff
@@ -2084,7 +2084,7 @@ and tc_abs env (top:term) (bs:binders) (body:term) : term * lcomp * guard_t =
       //we don't abstract over subtyping constraints; so solve them now
       //but leave out the tactics constraints for later so that the tactic
       //can have a more global view of all the constraints
-      let guard_body = Rel.solve_non_tactic_deferred_constraints envbody guard_body in
+      let guard_body = Rel.solve_non_tactic_deferred_constraints true envbody guard_body in
```

Comment:
When finishing the body of a lambda term, solve deferred constraints but allow deferring flex-flex